### PR TITLE
ff

### PR DESCRIPTION
**Pull request by prihit**

### CODE REVIEW

File: profiles/views/authentication.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from django.contrib.auth import authenticate, login, logout
 from django.shortcuts import redirect, render
 from django.views.generic.base import View
-
+from myCollegeAI import urls
```
Comment (chayan007) [inline profiles/views/authentication.py:4]: Remove this. This is class circular import issue !

File: profiles/views/authentication.py [hunk 2]
```diff
@@ -26,8 +26,10 @@ class RegisterView(View):
 
     def post(self, request):
         try:
-            first_name = request.POST['first_name']
-            last_name = request.POST['last_name']
+            name = request.POST['name']
+            name = name.split(" ",1)
```
Comment (chayan007) [inline profiles/views/authentication.py:30]: name.split()[1]

File: profiles/views/authentication.py [hunk 3]
```diff
@@ -26,8 +26,10 @@ class RegisterView(View):
...
+            first_name = name[0]
+            last_name = name[-1]
```
Comment (chayan007) [inline profiles/views/authentication.py:32]: Handle the case if a person has also entered middle name !

File: templates/v2/components/modals/register.html [hunk 4]
```diff
@@ -11,10 +11,10 @@ <h6 class="sub-header-03">Log In</h6>
       <div class="mobile_modal-switcher">
         <p class="link-01">Don't Have an Account?<a class="link-primary" href="#signup" id="signUpButton">Sign Up</a></p>
       </div>
-      <form class="form-signup" data-validate action="" method="">
+      <form class="form-signup" data-validate action="{% url 'login' %}"  method="POST">
```
Comment (chayan007) [inline templates/v2/components/modals/register.html:14]: remove data-validate

File: templates/v2/layouts/home.html [hunk 5]
```diff
@@ -33,10 +33,11 @@ <h6 class="sub-header-03">Log In</h6>
           <div class="mobile_modal-switcher">
             <p class="link-01">Don't Have an Account?<a class="link-primary" href="#signup" id="signUpButton">Sign Up</a></p>
           </div>
-          <form class="form-signup" data-validate action="" method="">
+          <form class="form-signup" data-validate action="{% url 'login' %}" method="POST">
```
Comment (chayan007) [inline templates/v2/layouts/home.html:36]: remove data-validate

File: profiles/views/authentication.py (hunk 2 above)
Comment (prihit) [reply profiles/views/authentication.py:30]: This is done for middle name. Like in firstname the first name is stored rest whole name is stored in the lastname.

File: templates/v2/components/modals/register.html (hunk 4 above)
Comment (prihit) [reply templates/v2/components/modals/register.html:14]: done. what about is_authenticated ?

File: profiles/views/authentication.py [hunk 6]
```diff
@@ -26,8 +26,10 @@ class RegisterView(View):
...
             username = request.POST['username']
```
Comment (chayan007) [inline profiles/views/authentication.py:33]: `username = request.POST['email']`

File: profiles/views/authentication.py (hunk 1 above)
Comment (chayan007) [reply profiles/views/authentication.py:4]: @prihit

File: templates/v2/layouts/home.html [hunk 7]
```diff
@@ -305,6 +316,12 @@ <h5>
     </section>
     {% block content %} {% endblock %}
   </div>
+ <div id="modalpsyid" class="modalpsy">
```
Comment (chayan007) [inline templates/v2/layouts/home.html:319]: What is this for ?

**Comment by chayan007** [comment]
Please do not forget to add the alert boxes in home page.

File: templates/v2/layouts/home.html (hunk 7 above)
Comment (prihit) [reply templates/v2/layouts/home.html:319]: this was for the modal for psychometric

**Review by chayan007 — APPROVED** [review]

**Pull request merged**